### PR TITLE
feat(controlplaneauth): add option to use token from robot token 

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
Give the choice to use a robot token for control plane authentication instead of making a personal access token. 
Just add a field to mention the robot token connectionSecret

make sure all the necessary resources like team, robot, and permissions are set up beforehand.

Fixes #

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

```
 kubectl get managed
NAME                                   READY   SYNCED   EXTERNAL-NAME                          AGE
robot.iam.upbound.io/serviceaccount2   True    True     eb31da85-1132-4982-9960-7467dfe4dde4   35m

NAME                                                        READY   SYNCED   EXTERNAL-NAME   AGE
robotteammembership.iam.upbound.io/serviceaccount-team-a2   True    True                     35m

NAME                          READY   SYNCED   EXTERNAL-NAME                          AGE
team.iam.upbound.io/team-a2   True    True     ffcf8166-3a7f-4b59-bde8-c24b59a780fa   35m

NAME                                 READY   SYNCED   EXTERNAL-NAME                          AGE
token.iam.upbound.io/access-token2   True    True     b916fb94-0b08-4ed0-bf0c-e8e1ad4defdc   35m

NAME                                                   READY   SYNCED   EXTERNAL-NAME          AGE
controlplaneauth.mcp.upbound.io/controlplane-a-token   True    True     controlplane-a-token   6m32s

NAME                                                 READY   SYNCED   EXTERNAL-NAME   AGE
controlplanepermission.mcp.upbound.io/owner-team-a   True    True                     34m
```

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: internal/controller/controlplaneauth/controller.go [hunk 1]
```diff
@@ -133,16 +133,25 @@ func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.Ex
...
-	uid, err := uuid.Parse(meta.GetExternalName(cr))
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(err, fmt.Sprintf("failed to parse external name as UUID %s", meta.GetExternalName(cr)))
-	}
-	_, err = c.tokens.Get(ctx, uid)
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(uperrors.IsNotFound, err), "failed to get token")
+
+	if cr.Spec.ForProvider.TokenSecretRef == nil {
+		uid, err := uuid.Parse(meta.GetExternalName(cr))
```
Comment (tnthornton) [inline internal/controller/controlplaneauth/controller.go:147]: FYI - we don't place guarantees internally on this ID being stable. i.e. if we do a restore of the control plane, the id will change.

File: internal/controller/controlplaneauth/controller.go [hunk 2]
```diff
@@ -160,29 +169,39 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
...
-				Data: tokens.TokenOwnerData{
-					Type: tokens.TokenOwnerUser,
-					ID:   userID,
+		t, err := c.tokens.Create(ctx, &tokens.TokenCreateParameters{
+			Attributes: tokens.TokenAttributes{
+				Name: fmt.Sprintf("%.64s", cr.Spec.ForProvider.ControlPlaneName+"-"+string(cr.UID)),
+			},
+			Relationships: tokens.TokenRelationships{
+				Owner: tokens.TokenOwner{
+					Data: tokens.TokenOwnerData{
```
Comment (tnthornton) [inline internal/controller/controlplaneauth/controller.go:184]: Nit/non-blocking the naming of these attributes stutters with the package name.

**Review by tnthornton — APPROVED** [review]

File: internal/controller/controlplaneauth/controller.go (hunk 1 above)
Comment (haarchri) [reply internal/controller/controlplaneauth/controller.go:147]: it's the uid from an PAT

Comment (tnthornton) [reply internal/controller/controlplaneauth/controller.go:147]: Ahh kk. 👍

**Pull request merged**